### DISCUSSION TITLE
NullSubstitute with dynamic value

### DISCUSSION THREAD

**Discussion by dnperfors** [LuckyPennySoftware/AutoMapper#3732] [NA]
Given the following types:
```
public record Source
{
    public DateTime? Date { get; init; }
}

public record Destination
{
    public DateTime Date { get; init; }
}
```
When I map from `Source` to `Destination`, I want to substitute the null value for Date to `DateTime.Today`, however it looks like that the `NullSubstitute` expects a fixed value. Would it be possible to get an overload of `NullSubstitute` that accepts a `Func<object>`?

**Comment by lbargaoanu** [comment]
You can easily use a resolver or a value converter for that.
